torch/csrc/jit/python/init.cpp

```diff
@@ -827,13 +827,34 @@ void initJITBindings(PyObject* module) {
 
   py::class_<PythonFutureWrapper, std::shared_ptr<PythonFutureWrapper>>(
       m, "Future")
+      .def(py::init([]() {
+        return std::make_shared<PythonFutureWrapper>(
+            c10::make_intrusive<c10::ivalue::Future>(PyObjectType::get()));
+      }))
       .def(
           "wait",
           &PythonFutureWrapper::wait,
           py::call_guard<py::gil_scoped_release>())
       .def(
-          "_then",
+          "then",
           &PythonFutureWrapper::then,
+          py::call_guard<py::gil_scoped_release>())
+      .def(
+          "set_result",
+          // Intentionally not releasing GIL
+          &PythonFutureWrapper::markCompleted)
+      .def(
```

Comment:
Should we use `py::pickle`, as in https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/rpc/init.cpp#L295, otherwise I think there will be a bunch of warnings spewed in logs?

Reply:
Yes, let me update. Thanks!